libs/wallet/src/connect-dialog/connect-dialog.tsx

```diff
@@ -1,82 +1,274 @@
+  Loader,
+} from '@vegaprotocol/ui-toolkit';
+import { useState } from 'react';
+import { t, useChainIdQuery } from '@vegaprotocol/react-helpers';
 import type { VegaConnector } from '../connectors';
-import { RestConnector } from '../connectors';
-import { RestConnectorForm } from '../rest-connector-form';
-import { useVegaWallet } from '../use-vega-wallet';
+import { JsonRpcConnector, RestConnector } from '../connectors';
+import { RestConnectorForm } from './rest-connector-form';
+import { JsonRpcConnectorForm } from './json-rpc-connector-form';
+import { Networks, useEnvironment } from '@vegaprotocol/environment';
+import {
+  ConnectDialogContent,
+  ConnectDialogFooter,
+  ConnectDialogTitle,
+} from './connect-dialog-elements';
+
+type Connectors = { [key: string]: VegaConnector };
+type WalletType = 'gui' | 'cli' | 'hosted';
 
 export interface VegaConnectDialogProps {
-  connectors: { [name: string]: VegaConnector };
+  connectors: Connectors;
   dialogOpen: boolean;
   setDialogOpen: (isOpen: boolean) => void;
 }
 
-export function VegaConnectDialog({
+export const VegaConnectDialog = ({
   connectors,
   dialogOpen,
   setDialogOpen,
-}: VegaConnectDialogProps) {
-  const { connect } = useVegaWallet();
-
-  // Selected connector, we need to show the auth form if the rest connector (which is
-  // currently the only way to connect) is selected.
-  const [selectedConnector, setSelectedConnector] =
-    useState<VegaConnector | null>(null);
+}: VegaConnectDialogProps) => {
+  const { data } = useChainIdQuery();
 
-  // Connects with the provided connector instance and closes the modal
-  const connectAndClose = useCallback(
-    (connector: VegaConnector) => {
-      connect(connector);
-      setDialogOpen(false);
-    },
-    [connect, setDialogOpen]
+  return (
+    <Dialog open={dialogOpen} size="small" onChange={setDialogOpen}>
+      <ConnectDialogContainer
+        connectors={connectors}
+        closeDialog={() => setDialogOpen(false)}
+        appChainId={data?.statistics.chainId}
+      />
+    </Dialog>
   );
+};
+
+const ConnectDialogContainer = ({
+  connectors,
+  closeDialog,
+  appChainId,
+}: {
+  connectors: Connectors;
+  closeDialog: () => void;
+  appChainId?: string;
+}) => {
+  const { VEGA_WALLET_URL, VEGA_ENV, HOSTED_WALLET_URL } = useEnvironment();
+  const [selectedConnector, setSelectedConnector] = useState<VegaConnector>();
+  const [walletUrl, setWalletUrl] = useState(VEGA_WALLET_URL || '');
+  const [walletType, setWalletType] = useState<WalletType>();
 
-  // Effect to immediately connect if the selected connector is NOT a
-  // rest connector
-  useEffect(() => {
-    if (
-      selectedConnector !== null &&
-      !(selectedConnector instanceof RestConnector)
-    ) {
-      connectAndClose(selectedConnector);
-    }
-  }, [selectedConnector, connectAndClose]);
+  if (!appChainId) {
+    return (
+      <ConnectDialogContent>
+        <ConnectDialogTitle>{t('Fetching chain ID')}</ConnectDialogTitle>
```

Comment:
suggestion: wouldn't it be better rephrased to "Fetching network"?

Reply:
Up for debate. This was suggested by ux